qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/FaissSQIT.java

```diff
@@ -0,0 +1,273 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.bwc;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.opensearch.client.Response;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.knn.KNNResult;
+import org.opensearch.knn.common.KNNConstants;
+import org.opensearch.knn.index.KNNMethod;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.knn.index.query.KNNQueryBuilder;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.TreeMap;
+
+import static org.opensearch.knn.common.KNNConstants.ENCODER_SQ;
+import static org.opensearch.knn.common.KNNConstants.FAISS_NAME;
+import static org.opensearch.knn.common.KNNConstants.FAISS_SQ_CLIP;
+import static org.opensearch.knn.common.KNNConstants.FAISS_SQ_ENCODER_FP16;
+import static org.opensearch.knn.common.KNNConstants.FAISS_SQ_TYPE;
+import static org.opensearch.knn.common.KNNConstants.KNN_ENGINE;
+import static org.opensearch.knn.common.KNNConstants.METHOD_ENCODER_PARAMETER;
+import static org.opensearch.knn.common.KNNConstants.METHOD_IVF;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_SPACE_TYPE;
+import static org.opensearch.knn.common.KNNConstants.MODEL_ID;
+import static org.opensearch.knn.common.KNNConstants.PARAMETERS;
+import static org.opensearch.knn.common.KNNConstants.NAME;
+
+public class FaissSQIT extends AbstractRestartUpgradeTestCase {
+    private static final String TEST_FIELD = "test-field";
+    private static final String TRAIN_TEST_FIELD = "train-test-field";
+    private static final String TRAIN_INDEX = "train-index";
+    private static final String TEST_MODEL = "test-model";
+
+    public void testHNSWSQFP16_onUpgradeWhenIndexedAndQueried_thenSucceed() throws Exception {
+        if (!isRunningAgainstOldCluster()) {
+            KNNMethod hnswMethod = KNNEngine.FAISS.getMethod(KNNConstants.METHOD_HNSW);
+            SpaceType[] spaceTypes = { SpaceType.L2, SpaceType.INNER_PRODUCT };
+            Random random = new Random();
+            SpaceType spaceType = spaceTypes[random.nextInt(spaceTypes.length)];
+
+            List<Integer> mValues = ImmutableList.of(16, 32, 64, 128);
+            List<Integer> efConstructionValues = ImmutableList.of(16, 32, 64, 128);
+            List<Integer> efSearchValues = ImmutableList.of(16, 32, 64, 128);
+
+            int dimension = 128;
+            int numDocs = 100;
+
+            // Create an index
```

Comment:
can you add json string for below builder inside comment? Thanks

Reply:
updated